src/RamseyCacheKeyGenerator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Library\HealthCheckCache;
+
+use Ramsey\Uuid\Uuid;
+
+final class RamseyCacheKeyGenerator implements CacheKeyGeneratorInterface
+{
+    private const DEFAULT_PREFIX = 'oat-health-check';
+
+    /** @var string */
+    private $prefix;
+
+    public function __construct(string $prefix = null)
+    {
+        $this->prefix = $prefix ?? self::DEFAULT_PREFIX;
+    }
+
+    public function generate(): string
```

Comment:
actually, what about allowing to pass the prefix (default = constant) at `generate()` instead of `__construct()` ? this would make this class even more usable, wdyt ?

Reply:
optional: if you do so, you could also move the default prefix constant to the CacheKeyGeneratorInterface (draws stronger contract for implementations)

Reply:
not sure I get it. what is(are) the other reason(s) you would use this class for?

Reply:
no I meant if you pass the prefix a generation time (in generate()) instead of at construction time, you make this generator easier to use.

=> this becomes particulary handy when this generator will be used as cached service in a container (like in many major PHP framworks), you wont need to tweak the container to be able to generate keys with different prefixes :) you'll be able to change the prefix, if you want, when you'll call generate() all simply

Reply:
Yes, that's true, usage together with a container can become easier that way. But that raises another complication: we will have to add prefix parameter to the `CacheKeyGeneratorInterface::generate()` method too. Wouldn't that give too much force for any new implementation? What if a developer does not want to use any prefix in their implementation at all?

Reply:
we could make it nullable.

but anyway lets stop splitting hairs and release this way, we will see at usage if we need to move from constructor to generate() and if needed we can make another release